tests/Api/Category/ListCategoryIntegration.php

```diff
@@ -0,0 +1,164 @@
+
+        $this->assertSame($uriGenerator->generate(CategoryApi::CATEGORIES_PATH, [], [
+            'page' => 2,
+            'limit' => 2,
+            'with_count' => false,
+        ]), $firstPage->getNextLink());
+
+        $categories = $firstPage->getItems();
+        $this->assertCount(2 ,$categories);
+        $this->assertSameContent($expectedCategories[0], $categories[0]);
+        $this->assertSameContent($expectedCategories[1], $categories[1]);
+
+        $secondPage = $firstPage->getNextPage();
+        $this->assertInstanceOf(PageInterface::class, $secondPage);
+        $this->assertSame($uriGenerator->generate(CategoryApi::CATEGORIES_PATH, [], [
+            'page' => 1,
+            'limit' => 2,
+            'with_count' => false,
+        ]), $secondPage->getPreviousLink());
+
+        $categories = $secondPage->getItems();
+        $this->assertCount(2 ,$categories);
+        $this->assertSameContent($expectedCategories[2], $categories[0]);
+        $this->assertSameContent($expectedCategories[3], $categories[1]);
+
+        $lastPage = $secondPage->getNextPage();
+        $this->assertInstanceOf(PageInterface::class, $lastPage);
+        $this->assertNull($lastPage->getNextPage());
+
+        $categories = $lastPage->getItems();
+        $this->assertCount(1 ,$categories);
+        $this->assertSameContent($expectedCategories[4], $categories[0]);
+
+        $previousPage = $lastPage->getPreviousPage();
+        $this->assertInstanceOf(PageInterface::class, $previousPage);
+
+        $categories = $previousPage->getItems();
+        $this->assertCount(2 ,$categories);
+        $this->assertSameContent($expectedCategories[2], $categories[0]);
+        $this->assertSameContent($expectedCategories[3], $categories[1]);
+    }
+
+    public function testListPerPageWithCount()
+    {
+        $api = $this->createClient()->getCategoryApi();
+
+        $firstPage = $api->listPerPage(2, true);
+        $this->assertInstanceOf(PageInterface::class, $firstPage);
+        $this->assertSame(5, $firstPage->getCount());
+    }
+
+    public function testAll()
+    {
+        $api = $this->createClient()->getCategoryApi();
+        $categories = $api->all();
+        $expectedCategories = $this->getExpectedCategories();
+
+        $this->assertInstanceOf(ResourceCursorInterface::class, $categories);
+
+        $categoriesCount = 0;
+        foreach ($categories as $key => $category) {
+            $this->assertInternalType('array', $category);
+            $this->assertSameContent($expectedCategories[$key], $category);
+            $categoriesCount++;
+        }
+
+        $this->assertSame(5, $categoriesCount);
+    }
+
+    protected function getExpectedCategories()
+    {
+        $uriGenerator = $this->getUriGenerator();
+
+        return [
+            [
+                '_links' => [
+                    'self' => [
+                        'href' => $uriGenerator->generate(CategoryApi::CATEGORY_PATH, ['code' => '2014_collection']),
```

Comment:
same comment as above, avoid to call service to generate your route. hardcode them please